src/core/server/saved_objects/saved_objects_service.ts

```diff
@@ -19,26 +19,37 @@
 
 import { CoreService } from 'src/core/types';
 import { first } from 'rxjs/operators';
+import {
+  SavedObjectsClient,
+  SavedObjectsSchema,
+  SavedObjectsRepository,
+  SavedObjectsSerializer,
+  ScopedSavedObjectsClientProvider,
+} from './';
+import { getRootPropertiesObjects } from './mappings';
 import { KibanaMigrator, IKibanaMigrator } from './migrations';
 import { CoreContext } from '../core_context';
 import { LegacyServiceSetup } from '../legacy/legacy_service';
 import { ElasticsearchServiceSetup } from '../elasticsearch';
 import { KibanaConfigType } from '../kibana_config';
 import { retryCallCluster } from '../elasticsearch/retry_call_cluster';
 import { SavedObjectsConfigType } from './saved_objects_config';
+import { KibanaRequest } from '../http';
 import { Logger } from '..';
 
 /**
  * @public
  */
-// eslint-disable-next-line @typescript-eslint/no-empty-interface
-export interface SavedObjectsServiceSetup {}
+export interface SavedObjectsServiceSetup {
+  clientProvider: Pick<ScopedSavedObjectsClientProvider, keyof ScopedSavedObjectsClientProvider>;
```

Comment:
Should this be available in setup?

Reply:
This is so that we can call `registerCoreContext` from setup which calls `http.registerRouteHandlerContext` https://github.com/skaapgif/kibana/blob/08e8a405deff55dff474d87dbcc530d0c50e5b6b/src/core/server/server.ts#L148-L149

Reply:
What about other core services that want to use scoped clients? Are they allowed to use context only? That adds a bit of magic because a dependent service needn't declare a dependency on the other service. 
Also we may need this for LP parts having no access to a context.

Reply:
I agree. In legacy we only exposed a scoped saved objects client as "standard". But some plugins ended up having to create their own saved objects repository and clients in order to use the internal user, so it would be better to expose this. Because this is blocking so many people and I'd like to do that in a second PR though.